### PR TITLE
uploading homework lesson 13

### PR DESCRIPTION
**Pull request by edisabou**

### CODE REVIEW

File: lesson_13/edmond_sabou/home1_les13.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from math import sqrt
+
+
+class Perimeter:
```
Comment (Liitle) [inline lesson_13/edmond_sabou/home1_les13.py]: For Mixin classes, we always add the word `Mixin` as a suffix

File: lesson_13/edmond_sabou/home1_les13.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def perimeter(self):
+        class_name = type(self).__name__
+        return f'The perimeter for {class_name} is {sum(self.sides)}.\n'
+
+
+class Polygon(Perimeter):
```
Comment (Liitle) [inline lesson_13/edmond_sabou/home1_les13.py]: Placing the Mixin here will make the perimeter available for all shapes that inherit from Polygon.

Where we can move the mixin to have the perimeter method only on instances of Triangles?

File: lesson_13/edmond_sabou/home1_les13.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+class Square(Polygon):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        class_name = type(self).__name__
+        side, *_ = self.sides
+        return f"The {class_name}'s area is {side ** 2}."
+
+    def from_area(self):
```
Comment (Liitle) [inline lesson_13/edmond_sabou/home1_les13.py]: This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create a new object we need the class object (cls).

What we can use from python to achieve this?

**Review by Liitle — CHANGES_REQUESTED** [review]

**Comment by Liitle** [comment]
Hi @edisabou, can you please add the changes. I want to have every PR merged before the midterm assignment. 

If I can assist with anything let me know :smile:

File: lesson_13/edmond_sabou/home1_les13.py (hunk 1 above)
Comment (edisabou) [reply lesson_13/edmond_sabou/home1_les13.py]: added suffix

File: lesson_13/edmond_sabou/home1_les13.py (hunk 2 above)
Comment (edisabou) [reply lesson_13/edmond_sabou/home1_les13.py]: moved the Mixin just above the Triangles class, moved every other class above the Mixin class

File: lesson_13/edmond_sabou/home1_les13.py (hunk 3 above)
Comment (edisabou) [reply lesson_13/edmond_sabou/home1_les13.py]: changed the method to 
```
@classmethod
    def from_area(cls, sq):
        sq = sqrt(sq)
        return sq
```

File: lesson_13/edmond_sabou/home1_les13.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    def area(self):
+        class_name = type(self).__name__
+        side, *_ = self.sides
+        return side ** 2
+
+    @classmethod
+    def from_area(cls, sq):
+        sq = sqrt(sq)
+        return sq
```
Comment (Liitle) [inline lesson_13/edmond_sabou/home1_les13.py]: Your class method returns `sq`.  In this case `sq` is a number, not an instance of Square. To have an alternative constructor you need to return an instance of the class. You have to options cls(sq,sq,sq,sq) or Square(sq,sq,sq,sq)

**Review by Liitle — CHANGES_REQUESTED** [review]

**Pull request merged**